inferelator_ng/tfa.py

```diff
@@ -0,0 +1,66 @@
+import numpy as np
+import pandas as pd
+from scipy import linalg
+
+class TFA:
+
+    """
+    TFA calculates transcription factor activity using matrix pseudoinverse
+
+        Parameters
+    --------
+    prior: pd.dataframe
+        binary g by t matrix stating existence of gene-TF interactions. 
+        g--gene, t--TF.
+
+    exp.mat: pd.dataframe
+        normalized expression matrix.
+
+    exp.mat.halftau: pd.dataframe
+        normalized expression matrix for time series.
+
+    noself=True: boolean
+        By default, self-regulatory interactions are removed.
+
+    dup_self=True: boolean
+        By default, self interactions for TFs that other TFs with the 
+        exact same set of interactions in the prior are kept
+    """
+
+    def __init__(self, prior, exp_mat, exp_mat_halftau):
+        self.prior = prior
+        self.exp_mat = exp_mat
+        self.exp_mat = exp_mat_halftau
+
+    def tfa(self, noself = True, dup_self = True):
+        tfwt = self.prior.abs().sum(axis = 0) > 0
```

Comment:
It's not immediately clear what `tfwt` represents, especially if your are not fluent in numpy conventions.
I'd like better names or comments explaining the computations and intension, please.